hack/e2e/run.sh

```diff
@@ -133,6 +133,17 @@ else
   else
     set -x
     set +e
+    # Fail early if running single-az tests AND current cluster has nodes in multiple AZs. TODO see if we can add node selector for single-az tests.
+    if [[ "$GINKGO_FOCUS" =~ "single-az" ]]; then
+      if [[ $(kubectl get nodes \
+        --selector '!node-role.kubernetes.io/control-plane' \
+        -o jsonpath='{.items[*].metadata.labels.topology\.kubernetes\.io/zone}' |
+        sort | uniq | wc -l) -ne 1 ]]; then
```

Comment:
Can we do greater than 1 (rather than not equal), so if the nodes don't have the topology label for whatever reason this doesn't break that case?